lib/migration.pm

```diff
@@ -174,7 +174,10 @@ sub check_rollback_system {
         base_version=\$(echo \$version | cut -d'-' -f1)
         zypper lr | cut -d'|' -f3 | gawk '/SLE/ || /openSUSE/' | sed \"/\$version\\|Module.*\$base_version/d\"
     ", 100);
-    record_info('Incorrect Repos', $incorrect_repos, result => 'fail') if $incorrect_repos;
+    # for leap to sle migration, we'll have SLE-$VERSION-Updates in the leap repo.
+    if ($incorrect_repos) {
+        record_info('Incorrect Repos', $incorrect_repos, result => 'fail') unless (is_leap_migration && $incorrect_repos eq 'SLE-' . get_var("VERSION") . '-Updates');
```

Comment:
I prefer to move `$incorrect_repos eq 'SLE-' . get_var("VERSION") . '-Updates'` to previous `if`. like this
`if ($incorrect_repos && $incorrect_repos ne 'SLE-' . get_var("VERSION") . '-Updates')`
Maybe you could try to combine the `if` and `unless`.

Reply:
we'd check is_leap_migration for that. it's twisted mind-blowing, I just wan to make it
easy and clean for me.